packages/lib/src/toolbar/floating/ResetZoomButton.tsx

```diff
@@ -1,21 +1,16 @@
-import { useFrame, useThree } from '@react-three/fiber';
-import { useState } from 'react';
+import { useThree } from '@react-three/fiber';
 
+import { useCameraState } from '../../vis/hooks';
 import FloatingControl from './FloatingControl';
 import styles from './ResetZoomButton.module.css';
 
 function ResetZoomButton() {
   const camera = useThree((state) => state.camera);
   const invalidate = useThree((state) => state.invalidate);
 
-  const [isVisible, setVisible] = useState(false);
-
-  useFrame(() => {
-    const isZoomedIn = camera.scale.x < 1 || camera.scale.y < 1;
-    if (isVisible !== isZoomedIn) {
-      setVisible(isZoomedIn);
-    }
-  });
+  const isVisible = useCameraState(({ scale }) => {
+    return scale.x < 1 || scale.y < 1;
+  }, []);
```

Comment:
Here, `useCameraState` makes sense, since we need to update the DOM, so I'm refactoring the code to actually use it. The behaviour is the same, just shorter code.

Reply:
Nice. I would keep `isZoomedIn` as the variable name, though.